guides/elixir.md

```diff
@@ -0,0 +1,937 @@
+  not File.exists?(path)
+  ```
+
+* <a name="default-arguments"></a>
+  Use spaces around default arguments `\\` definition.
+  <sup>[[link](#default-arguments)]</sup>
+
+  ```elixir
+  # Bad
+  def start_link(fun, options\\[])
+
+  # Good
+  def start_link(fun, options \\ [])
+  ```
+
+* <a name="bitstring-segment-options"></a>
+  Do not put spaces around segment options definition in bitstrings.
+  <sup>[[link](#bitstring-segment-options)]</sup>
+
+  ```elixir
+  # Bad
+  <<102 :: unsigned-big-integer, rest :: binary>>
+  <<102::unsigned - big - integer, rest::binary>>
+
+  # Good
+  <<102::unsigned-big-integer, rest::binary>>
+  ```
+
+* <a name="leading-space-comment"></a>
+  Use one space between the leading `#` character of the comment and the text of the comment.
+  <sup>[[link](#leading-space-comment)]</sup>
+
+  ```elixir
+  # Bad
+  #Amount to take is greater than the number of elements
+
+  # Good
+  # Amount to take is greater than the number of elements
+  ```
+
+* <a name="space-before-anonymous-fun-arrow"></a>
+  Always use a space before `->` in 0-arity anonymous functions.
+  <sup>[[link](#space-before-anonymous-fun-arrow)]</sup>
+
+  ```elixir
+  # Bad
+  Task.async(fn->
+    ExUnit.Diff.script(left, right)
+  end)
+
+  # Good
+  Task.async(fn ->
+    ExUnit.Diff.script(left, right)
+  end)
+  ```
+
+### Indentation
+
+* <a name="binary-ops-indentation"></a>
+  <a name="guard-clauses"></a>
+  Indent the right-hand side of a binary operator one level more than the left-hand side if left-hand side and right-hand side are on different lines. The only exceptions are `when` in guards and `|>`, which go on the beginning of the line and should be indented at the same level as their left-hand side. Do this also for binary operators when assigning.
+  <sup>[[link](#binary-ops-indentation)]</sup>
+
+  ```elixir
+  # Bad
+
+  "No matching message.\n" <>
+  "Process mailbox:\n" <>
+  mailbox
+
+  message =
+    "No matching message.\n" <>
+    "Process mailbox:\n" <>
+    mailbox
+
+  input
+    |> String.strip()
+    |> String.downcase()
+
+  defp valid_identifier_char?(char)
+    when char in ?a..?z
+      when char in ?A..?Z
+      when char in ?0..?9
+      when char == ?_ do
+    true
+  end
+
+  defp parenless_capture?({op, _meta, _args})
+       when is_atom(op) and
+       atom not in @unary_ops and
+       atom not in @binary_ops do
+    true
+  end
+
+  # Good
+
+  "No matching message.\n" <>
+    "Process mailbox:\n" <>
+    mailbox
+
+  message =
+    "No matching message.\n" <>
+      "Process mailbox:\n" <>
+      mailbox
+
+  input
+  |> String.strip()
+  |> String.downcase()
+
+  defp valid_identifier_char?(char)
+       when char in ?a..?z
+       when char in ?A..?Z
+       when char in ?0..?9
+       when char == ?_ do
+    true
+  end
+
+  defp parenless_capture?({op, _meta, _args})
+       when is_atom(op) and
+              atom not in @unary_ops and
+              atom not in @binary_ops do
+    true
+  end
```

Comment:
Fix indentation for these two?

Reply:
I agree, this looks ugly.
I was trying to understand why and seems: 
```elixir
defp parenless_capture?({op, _meta, _args})
     when is_atom(op) and
            atom not in @unary_ops and
            atom not in @binary_ops do
```
This all is just a "head" of function (from `def` to `do`)... in a one line it will be very long
```elixir
defp parenless_capture?({op, _meta, _args}) when is_atom(op) and atom not in @unary_ops and atom not in @binary_ops do
```
so we break the "head" into multiple lines and to separate "visually" from the "body".. 
since body is "2 spaces" indented - we align head with the name of the function, so in case of private function `defp` it's 5, and in case of `def` it's 4 respectively. The rest of a "broken guards" are indented further... 

yeah.. it's opinionated, BUT in the end of a day that's what `mix format` does. 
> Elixir v1.6 introduced a [Code Formatter](https://hexdocs.pm/elixir/Code.html#format_string!/2) and [Mix format](https://hexdocs.pm/mix/Mix.Tasks.Format.html) task.
 The formatter should be preferred for all new projects and source code.